orbit-android/src/main/java/com/babylon/orbit/internal/AutoDispose.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 Babylon Partners Limited
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.babylon.orbit.internal
+
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import io.reactivex.disposables.Disposable
+
+internal fun Disposable.autoDispose(lifecycleOwner: LifecycleOwner) {
```

Comment:
Cool.

It's super important for us to get this right as this will affect every user of the library. Could you please add unit tests for every combination of lifecycle events here?

You can copy over most of the existing view model tests. Then I'd remove the view model tests and replace them with a simple sanity check making sure that we're setting up the lifecycle observer.

Reply:
Also could we please rename this to avoid any sort of confusion with the autodispose library?

`bindToLifecycle` could work